### ISSUE TITLE
Subtitle file gets overwritten during import. 

### ISSUE THREAD

**Issue by SugarDroid** [Radarr/Radarr#5378]
Hi there, just started using/testing Radarr v3 a few days ago.
This is related to #4646 but the workaround just made a new (imo worse) bug. 

**Describe the bug**
Many movies i get has forced subtitle and i want only that to be imported, but it gets overwritten during import. 
Meanwhile Radarr's GUI shows both imports like if there are 2 subtitles with same name, but only 1 file exist.

**To Reproduce**
Import a movie with the structure like this.

Movie/
├── movie.ext
├── movie.nfo
├── movie.srt                 <-- this is the forced subtitle
├── Sample
│   └── sample.ext
└── Sub
      └── movie.srt            <-- this is the full subtitle

**Expected behavior**
Only the forced subtitle gets imported from the top dir, and don't get overwritten by the full subtitle from subdir.

**Platform Information (please complete the following information):**
Version 3.0.0.4107
Package Version 3.0.0.4107-ls1 by linuxserver.io
.NET Core (3.1.10)
Docker 
Mode Console

**Comment by bakerboy448** [comment]
Logs...?

**Comment by austinwbest** [comment]
Not sure they would be needed. If order is to import root then look for subs in a subfolder, it would overwrite the one that was in root with the other.

Just need to check if one was imported from root & skip the one in a subfolder if that is the case I think. Prioritize the first one so to speak.

Q can confirm that

**Comment by SugarDroid** [comment]
@bakerboy448  https://pastebin.com/00SrwYsG

**Comment by Krandor1** [comment]
But what if I want the full subtitles in the sub folder?

Sounds like a toggle “search sub folders for subtitles” may be the way to go.

**Comment by SugarDroid** [comment]
@Qstick can we get the 1ebb71d reverted back until a proper solution for both use cases?

**Comment by austinwbest** [comment]
Reverting it isn't the solution, skipping existing I believe would be

**Comment by SugarDroid** [comment]
The problem is that until it's not reverted or properly fixed i can't import any movie which has subtitle, or have to manually fix every subtitles.

**Comment by JBou** [comment]
> But what if I want the full subtitles in the sub folder? Sounds like a toggle “search sub folders for subtitles” may be the way to go.

And what if I would like to keep **all** subtitle files (forced and full). Overriding or skipping, both doesn't work then 🤔

**Comment by austinwbest** [comment]
There can be only one {movie}.srt so unless numerical indexing is supported by media clients (I don't use them enough to know) then it is one or the other. Root folder would be priotized unless Q is willing to add a setting to pick when there is a conflict.

**Comment by SugarDroid** [comment]
> > But what if I want the full subtitles in the sub folder? Sounds like a toggle “search sub folders for subtitles” may be the way to go.
> 
> And what if I would like to keep **all** subtitle files (forced and full). Overriding or skipping, both doesn't work then

@JBou there's a possible way to keep them both, but that might can be a bit complex to accomplish. (I think that way could be the most optimal solution for all use cases.)

> There can be only one {movie}.srt so unless numerical indexing is supported by media clients (I don't use them enough to know) then it is one or the other. Root folder would be priotized unless Q is willing to add a setting to pick when there is a conflict.

@austinwbest Plex, Emby and also Kodi supports multiple subtitle files for the same movie, by naming them differently.
Like this example from [Plex's wiki](https://support.plex.tv/articles/200471133-adding-local-subtitles-to-your-media/):

/Movies
   /Avatar (2009)
         Avatar (2009).mkv
         Avatar (2009).eng.srt
         Avatar (2009).en.forced.ass
         Avatar (2009).en.sdh.srt
         Avatar (2009).de.srt
         Avatar (2009).de.sdh.forced.srt

Another example from [Emby's wiki](https://support.emby.media/support/solutions/articles/44001159160-subtitles):

/Movies
   /Home Alone (1990)
         Home Alone.mkv
         Home Alone.srt
         Home Alone.spa.forced.srt 
         Home Alone.spa.foreign.srt

And one more from [Kodi's wiki](https://kodi.wiki/view/Subtitles):

- Movie Name (2006).English.Forced.srt
- Movie Name (2006).en.forced.srt
- Movie Name (2006).German.Forced.srt
- Movie Name (2006)-Swedish-Forced.srt

As we can see all 3 servers supports both "language code" and "forced" flags in subtitle's file name. So, for the -totally legit- use case what @JBou mentioned, Radarr should compare their size and naming the small as movie.forced.srt and the big as movie.srt. If the proper -2 or 3 letters ISO- language code already exist in the source files then movie.lngcode.forced.srt and movie.lngcode.srt.

**Comment by austinwbest** [comment]
That is what I was curious about. So yea, adding forced wild be the solution. Is it 100% always smaller is forced? Is it 100% they won't be the same size? Again, I don't use subs so have to ask

**Comment by SugarDroid** [comment]
> That is what I was curious about. So yea, adding forced would be the solution. Is it 100% always smaller is forced? Is it 100% they won't be the same size? Again, I don't use subs so have to ask

@austinwbest forced subtitle is used to translate things that don't get dubbed translation. Something like:
- text on a billboard,
- article's title in a newspaper,
- book's title,
- name of places -like x y camp, country, headquarter-,
- lyrics when song is kept in its original language,
- someone talking different language than movie's default language.

A great example is [in this short clip](https://youtu.be/93DzOXD_91o). Without the forced subtitle you could have zero clue what he's talking about. :) (If you watch it, i think you'll realize that you too have subtitled movies, just they're hardcoded -like in the clip- and not in separated files.) 

On the other hand full subtitle appears during whole movie, even if you understand the spoken language.

I think everyone can agree in these two things as facts:
1. By their different purpose there's zero chance to have same size for both full and forced subtitle.
2. By its purpose forced subtitle can be big -depends on movie itself-, but always smaller than full subtitle.

Is it 100% that smaller is forced, technically not sure but i think we can take it up to 99.9%.  It's because there's the commentary subtitle which technically can be smaller than full, but never seen any movie which had _external_ commentary subtitle.
Then there's subtitle which can be even bigger than full, that's what SDH -“subtitles for the deaf and hard of hearing”-flag is used for, in Plex's example. (Plex supports also CC -closed captioning- flag.) Those subtitles has extra information -like applause, phone ring, thunder, etc- which is not needed for those who can hear them. Again, never seen any movie which had _external_ SDH (or CC) subtitle.

**Comment by Qstick** [comment]
Yea I think the solution here is improving our sub handling and parsing, we have similar issues with video file extras... Extras in general need some tlc.

**Comment by horstepipe** [comment]
hello
is there any update about this?
I'm just starting using radarr and now I'm having the same issue as the op. I'd like to keep both, full and forced subtitles. But this doesn't seem to be possible with Radarr?
I guess a workaround would be to directly mux all subtitles in a postprocessing script in sabnzbd if that's possible? Has anybody a script for idx/sub subtitles ready?

**Comment by D3nn3** [comment]
Unfortunately still an issue 😢

**Comment by bakerboy448** [comment]
Yes it's still an issue as it remains open and nothing has taken place to fix it.

No need to say it's an issue; we're aware.

**Comment by austinwbest** [comment]
> Unfortunately still an issue 😢 
> 
> Even having a pre-import-cleaning script naming everyhting nicely (e.g. `movie.en.srt` and `movie.en.forced.srt`) doesn't help.

Pre isn't going to really help, you need to do a post action if anything so it happens after radarr has moved in the files and does its thing (for now)
